.github/workflows/publish-to-pypi.yml

```diff
@@ -74,11 +71,54 @@ jobs:
         echo "Generated files:"
         ls -lh dist/
 
-    - name: Publish to Test PyPI
+    - name: Store the distribution packages
+      uses: actions/upload-artifact@v4.5.0
+      with:
+        name: python-package-distributions
+        path: dist/
+
+  publish-to-testpypi:
```

Comment:
Note that this testpypi step will likely fail for the backport PRs, due to version numbering conflicts with those on the main branch.